benchmark/sparse_blas/operations.cpp

```diff
@@ -566,6 +570,114 @@ class LookupOperation : public BenchmarkOperation {
+    const auto in_cols = host_input->get_const_col_idxs();
+    const auto factor_row_ptrs = host_factors->get_const_row_ptrs();
+    const auto factor_cols = host_factors->get_const_col_idxs();
+    std::unordered_set<itype> columns;
+    for (itype row = 0; row < num_rows; row++) {
+        const auto in_begin = in_cols + in_row_ptrs[row];
+        const auto in_end = in_cols + in_row_ptrs[row + 1];
+        const auto factor_begin = factor_cols + factor_row_ptrs[row];
+        const auto factor_end = factor_cols + factor_row_ptrs[row + 1];
+        columns.clear();
+        // the factor needs to contain the original matrix
+        // plus the diagonal if that was missing
+        columns.insert(in_begin, in_end);
+        columns.insert(row);
+        for (auto col_it = factor_begin; col_it < factor_end; ++col_it) {
+            const auto col = *col_it;
+            if (col >= row) {
+                break;
+            }
+            const auto dep_begin = factor_cols + factor_row_ptrs[col];
+            const auto dep_end = factor_cols + factor_row_ptrs[col + 1];
+            // insert the upper triangular part of the row
+            const auto dep_diag = std::find(dep_begin, dep_end, col);
+            columns.insert(dep_diag, dep_end);
+        }
+        // the factor should contain exactly these columns, no more
+        if (factor_end - factor_begin != columns.size()) {
+            return false;
+        }
+        for (auto col_it = factor_begin; col_it < factor_end; ++col_it) {
+            if (columns.find(*col_it) == columns.end()) {
+                return false;
+            }
+        }
+    }
+    return true;
+}
+
+
+class SymbolicLuOperation : public BenchmarkOperation {
+public:
+    explicit SymbolicLuOperation(const Mtx* mtx) : mtx_{mtx}, result_{} {}
+
+    std::pair<bool, double> validate() const override
+    {
+        return std::make_pair(
+            validate_symbolic_factorization(mtx_, result_.get()), 0.0);
+    }
+
+    gko::size_type get_flops() const override { return 0; }
+
+    gko::size_type get_memory() const override { return 0; }
+
+    void run() override { gko::factorization::symbolic_lu(mtx_, result_); }
+
+    void write_stats(rapidjson::Value& object,
+                     rapidjson::MemoryPoolAllocator<>& allocator) override
+    {
+        add_or_set_member(object, "factor_nonzeros",
+                          result_->get_num_stored_elements(), allocator);
+    }
+
+private:
+    const Mtx* mtx_;
+    std::unique_ptr<Mtx> result_;
+};
+
+
+class SymbolicCholeskyOperation : public BenchmarkOperation {
+public:
+    explicit SymbolicCholeskyOperation(const Mtx* mtx) : mtx_{mtx}, result_{} {}
+
+    std::pair<bool, double> validate() const override
+    {
+        return std::make_pair(
+            validate_symbolic_factorization(mtx_, result_.get()), 0.0);
+    }
+
+    gko::size_type get_flops() const override { return 0; }
+
+    gko::size_type get_memory() const override { return 0; }
```

Comment:
could we get the memory?

Reply:
doesn't make much sense to compute a bandwidth here, since we use both CPU and GPU for different parts of the algorithm